src/status_im/contexts/wallet/rpc.cljs

```diff
@@ -0,0 +1,61 @@
+(ns status-im.contexts.wallet.rpc
+  (:require [oops.core :as oops]
+            [promesa.core :as promesa]
+            [status-im.common.json-rpc.events :as rpc-events]
+            [status-im.constants :as constants]
+            [utils.hex]
+            [utils.transforms :as transforms]))
+
+(defn build-transaction
+  [chain-id tx]
+  (promesa/let [res (rpc-events/call-async "wallet_buildTransaction" true chain-id tx)]
+    {:message-to-sign (oops/oget res :messageToSign)
+     :tx-args         (oops/oget res :txArgs)}))
+
+(defn build-raw-transaction
+  [chain-id tx-args signature]
+  (-> (rpc-events/call-async "wallet_buildRawTransaction"
+                             true
+                             chain-id
+                             (transforms/js-stringify tx-args 0)
+                             signature)
+      (promesa/then #(oops/oget % "rawTx"))))
+
+(defn hash-message-eip-191
+  [message]
+  (rpc-events/call-async "wallet_hashMessageEIP191" true message))
+
+(defn safe-sign-typed-data
+  [data address password chain-id legacy?]
+  (rpc-events/call-async "wallet_safeSignTypedDataForDApps"
+                         true
+                         data
+                         address
+                         password
+                         chain-id
+                         legacy?))
+
+(defn get-suggested-fees
+  [chain-id]
+  (-> (rpc-events/call-async "wallet_getSuggestedFees" true chain-id)
+      (promesa/then transforms/js->clj)))
+
+(defn send-transaction-with-signature
+  [chain-id tx-args signature]
+  (rpc-events/call-async "wallet_sendTransactionWithSignature"
+                         true
+                         chain-id
+                         constants/transaction-pending-type-wallet-connect-transfer
+                         (transforms/js-stringify tx-args 0)
+                         signature))
+
+(defn sign-message
+  [message address password]
+  (-> (rpc-events/call-async "wallet_signMessage"
+                             true
+                             message
+                             address
+                             password)
+      (promesa/then (fn [s]
+                      {:message   message
+                       :signature (utils.hex/normalize-hex s)}))))
```

Comment:
Seems like the this function now returns the map instead of just the signature, but the previous usages weren't updated. I checked the `personal_sign` flow with walletconnect and it's broken, same as the other ones I assume. Below you can see malli being useful in debugging this :fireworks: 

```clj
Invalid function arguments:
  [{:message "0xc83023ae632ed42ebd6617c2c315e2e8dc6662426afa7f2dc8ed8bbf3489342b",
    :signature "fc40862dfccec2acdd8754d3f3a9850e4eb367d12c71a89335604eb176582670368558dd6bf345a98b57c0549bc5a0d5b92b4d976caea8e650238e3fe5d3387c01"}]

Function Var:
  utils.hex/prefix-hex

Input Schema:
  [:cat :string]

Errors:
  {:in [0],
   :message "should be a string",
   :path [0],
   :schema :string,
   :value {:message "0xc83023ae632ed42ebd6617c2c315e2e8dc6662426afa7f2dc8ed8bbf3489342b",
           :signature "fc40862dfccec2acdd8754d3f3a9850e4eb367d12c71a89335604eb176582670368558dd6bf345a98b57c0549bc5a0d5b92b4d976caea8e650238e3fe5d3387c01"}}
```